Automattic/simplenote-macos
# Decoupling PublishViewController

### Details
In this PR we're properly encapsulating `PublishViewController`, and moving its View(s) to a standalone **nib**, setup with Autolayout.

@aerych May I trouble you yeeeet again?
Thanks in advance!!!

### Test
1. Add a new note
2. Press over the "Share Action" button (top right corner, third button)
3. Click over `Publish to the Web`

- [x] Verify the note is not published
- [ ] Verify that clicking `Publish` displays the `Publishing...` UI
- [ ] Verify that after a while (can be few seconds) the UI switches to the `Published` state
- [ ] Verify that clicking `Unpublish` causes the UI to display `Unpublishing...`
- [ ] Verify that shortly after, the popver is back to the Unpublished state
- [ ] Verify Dark / Light mode please!

**Note:** Lots and lots of improvement areas, just focusing (for now) on offloading the main (massive) nib!
### Release
These changes do not require release notes.
